radixdlt/src/main/java/com/radixdlt/consensus/liveness/PacemakerImpl.java

```diff
@@ -80,28 +83,21 @@ public View getCurrentView() {
 
 	@Override
 	public Optional<View> processNewView(NewView newView) {
+		Hash newViewId = new Hash(Hash.hash256(Longs.toByteArray(newView.getView().number())));
 		ECDSASignature signature = newView.getSignature().orElseThrow(() -> new IllegalArgumentException("new-view is missing signature"));
```

Comment:
Would be better, IMO, to use the interfaces here, and avoid casting.  Make `pendingNewViews`:
```
private final Map<View, Signatures> pendingNewViews = new HashMap<>();
```

...and then this code could be:
```
Hash newViewId = new Hash(Hash.hash256(Longs.toByteArray(newView.getView().number())));
Signature signature = newView.getSignature().orElseThrow(() -> new IllegalArgumentException("new-view is missing signature"));
Signatures signatures = this.pendingNewViews.getOrDefault(newView.getView(), DefaultSignatures.emptySignatures());
signatures = signatures.concatenate(newView.getAuthor(), signature);
```